android/java/org/chromium/chrome/browser/firstrun/WelcomeOnboardingActivity.java

```diff
@@ -251,82 +270,60 @@ private boolean isDefaultBrowser() {
         return BraveSetDefaultBrowserUtils.isBraveSetAsDefaultBrowser(this);
     }
 
-    private void startTimer(int delayMillis) {
-        new Handler().postDelayed(this::nextOnboardingStep, delayMillis);
-    }
-
-    ActivityResultLauncher<String> mRequestPermissionLauncher = registerForActivityResult(
-            new ActivityResultContracts.RequestPermission(), isGranted -> { startTimer(3000); });
+    ActivityResultLauncher<String> mRequestPermissionLauncher =
+            registerForActivityResult(
+                    new ActivityResultContracts.RequestPermission(),
+                    isGranted -> {
+                        nextOnboardingStep();
+                    });
 
     private void nextOnboardingStep() {
         if (isActivityFinishingOrDestroyed()) return;
 
         mCurrentStep++;
+        // Step 0: Handle default browser setup
         if (mCurrentStep == 0) {
-            showIntroPage();
-        } else if (mCurrentStep == 1) {
-            if (!isNewOnboardingEnabled()
-                    || !BraveSetDefaultBrowserUtils.supportsDefaultRoleManager()) {
+            // For devices that don't support role manager API, show browser selection page
+            if (!BraveSetDefaultBrowserUtils.supportsDefaultRoleManager()) {
+                mIvBrave.setVisibility(View.VISIBLE);
                 showBrowserSelectionPage();
-            } else if (!isDefaultBrowser()) {
+            }
+            // If Brave is not default browser, trigger default browser prompt
```

Comment:
Instead of this comment, I think it would be nicer to rename `setDefaultBrowserAndProceedToNextStep` to `triggerDefaultBrowserPrompt`. But feel free to ignore.